x/pki/keeper/approved_certificates.go

```diff
@@ -54,6 +74,26 @@ func (k Keeper) RemoveApprovedCertificates(
 	))
 }
 
+func (k Keeper) removeCertFromList(serialNumber string, certs *types.ApprovedCertificates) {
```

Comment:
We only have guarantees that the combination of issuer and serial number will be unique; there is no guarantee that the serial numbers for all approved certificates will be unique